From e1a6e6b467b87c330595e175cbb8f9b64c617052 Mon Sep 17 00:00:00 2001
From: David Korth <gerbilsoft@gerbilsoft.com>
Date: Fri, 8 Mar 2024 03:41:42 -0500
Subject: [PATCH] [libromdata] isRomSupported_static(): Don't assert if
 info->ext is nullptr.

This might happen if the file doesn't have a file extension.

On the other hand, *do* assert if info is nullptr.

TODO: Add that assertion to more isRomSupported_static() functions?
---
 src/libromdata/Console/ColecoVision.cpp  | 1 -
 src/libromdata/Console/Intellivision.cpp | 1 -
 src/libromdata/Media/ISO.cpp             | 4 ++--
 3 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/libromdata/Console/ColecoVision.cpp b/src/libromdata/Console/ColecoVision.cpp
index bb94f5af2..c7766ffcf 100644
--- a/src/libromdata/Console/ColecoVision.cpp
+++ b/src/libromdata/Console/ColecoVision.cpp
@@ -340,7 +340,6 @@ ColecoVision::ColecoVision(const IRpFilePtr &file)
 int ColecoVision::isRomSupported_static(const DetectInfo *info)
 {
 	assert(info != nullptr);
-	assert(info->ext != nullptr);
 	if (!info || !info->ext) {
 		// Needs the file extension...
 		return -1;
diff --git a/src/libromdata/Console/Intellivision.cpp b/src/libromdata/Console/Intellivision.cpp
index 761ff116d..786190d64 100644
--- a/src/libromdata/Console/Intellivision.cpp
+++ b/src/libromdata/Console/Intellivision.cpp
@@ -194,7 +194,6 @@ Intellivision::Intellivision(const IRpFilePtr &file)
 int Intellivision::isRomSupported_static(const DetectInfo *info)
 {
 	assert(info != nullptr);
-	assert(info->ext != nullptr);
 	if (!info || !info->ext) {
 		// Needs the file extension...
 		return -1;
diff --git a/src/libromdata/Media/ISO.cpp b/src/libromdata/Media/ISO.cpp
index d82db352a..41f2a9126 100644
--- a/src/libromdata/Media/ISO.cpp
+++ b/src/libromdata/Media/ISO.cpp
@@ -707,8 +707,8 @@ void ISO::addMetaData_PVD(RomMetaData *metaData, const struct _ISO_Primary_Volum
 int ISO::isRomSupported_static(const DetectInfo *info)
 {
 	// NOTE: Only checking for supported file extensions.
-	assert(info->ext != nullptr);
-	if (!info->ext) {
+	assert(info != nullptr);
+	if (!info || !info->ext) {
 		// No file extension specified...
 		return -1;
 	}
